OrbitCore/FramePointerValidator.cpp

```diff
@@ -16,11 +17,13 @@ FramePointerValidator::GetFpoFunctions(
   std::vector<std::shared_ptr<Function>> result;
 
   cs_mode mode = is_64_bit ? CS_MODE_64 : CS_MODE_32;
-  csh handle;
-  if (cs_open(CS_ARCH_X86, mode, &handle) != CS_ERR_OK) {
+  csh temp_handle;
+  if (cs_open(CS_ARCH_X86, mode, &temp_handle) != CS_ERR_OK) {
     ERROR("Unable to open capstone.");
     return {};
   }
+  OrbitBase::unique_resource handle{std::move(temp_handle),
```

Comment:
wouldn't be adding `cs_close(&handle);` at the end of the function and in the early return be sufficient?

Edit: Just saw your PR comment. If I am not wrong, it should not throw an exception, however the managed resource is much cleaner. Thanks for this.

Reply:
This was exactly what I thought. The unique_resource also helps in the future when someone might add a potentially throwing call.